File: src/algorithms/recursion/js/fibonacci.js
/**
 *
 * @param {*} num
 * @returns
 */

/*
 * fibonacci iterative
 */
const fibonacci = (num) => {
  let res = 0;
  for (let i = 0; i <= num; i++) {
    res += i;
  }
  return res;
};

fibonacci(10);

/**
 * fibonaci recursive
 */
const fibonacciRecursive = (num) => {
  if (num <= 2) {
    return 1
  }

  return  fibonacciRecursive(num-1) + fibonacciRecursive(num-2)

};

fibonacciRecursive(10);  //Ouput -> 55
 

File: src/algorithms/recursion/py/fibonacci.py
'''
Fibonnaci iterative
'''

def fibonnaci(num):
    if num <=2:
        return 1
    
    res = 1
    curr = 1
    prev = 0
    while res < num:
        next = prev + curr
        prev = curr
        curr = next 
        res +=1
    return res


fibonnaci(10)



'''
Fibonnaci recursive
'''

def fibonnaciRecursive(num):
    if num <= 1:
        return num
    return fibonnaci(num - 1) + fibonnaci(num -2)

fibonnaciRecursive(10)